resources/repo.rb

```diff
@@ -40,36 +39,41 @@ def params = new JsonSlurper().parseText(args)
   converge_if_changed do
     nexus_api 'create_or_update_repo' do
       # TODO: Current nexus3_api passes a string and text/plain, change this back to a Hash later
-      args textify(name: new_resource.repo_name, type: new_resource.repo_type)
+      args textify(name: new_resource.repo_name,
+                   type: new_resource.repo_type,
+                   config: new_resource.config if new_resource.key?('config'))
       action :create, :run
       endpoint new_resource.api_url
       username new_resource.api_user
       password new_resource.api_password
 
       content <<EOS
-    // TODO: document parameters passed to this script.
-    // name: repo name
-    // type: repo type (equal to recipe for now?)
     import groovy.json.JsonSlurper
     import groovy.json.JsonOutput
 
     def params = new JsonSlurper().parseText(args)
-    // TODO: add checks for missing parameters.
 
-    if (repository.repositoryManager.get(params.name) == nil) {
-        // create
-        def config = new Configuration(
+    def repo = repository.repositoryManager.get(params.name)
+    Configuration conf
+    if (repo == nil) { // create
+        conf = new Configuration(
             repositoryName: params.name,
-            recipeName: params.type
+            recipeName: params.type,
+            online: true
+            attributes: params.config?.attributes
         )
-        repository.repositoryManager.createRepository(config)
-    } else {
-        // update
-        config = repository.repositoryManager.get(params.name).getConfiguration()
+        repository.repositoryManager.createRepository(conf)
+    } else { // update
+        conf = repo.getConfiguration()
+        conf.setRepositoryName(params.name)
+        conf.setRecipeName(params.type)
+        conf.setOnline(params.online)
```

Comment:
Undeclared `online` params. Please add it as nexus_repo property, and use it in repo create method